scripts/rmc-regression.sh

```diff
@@ -20,6 +20,9 @@ check-cbmc-viewer-version.py --major 2 --minor 5
 # Formatting check
 ./x.py fmt --check
 
+# C formatting check
+$SCRIPT_DIR/run-clang-format.sh -d 
```

Comment:
Not for this PR, but should we keep all formats check here or move them to their own scripts folder/workflow (for instance, with the copyright check)? Of course, the regression workflow should depend on that one.

I am happy to have a discussion about this.

Reply:
I actually ended up needing to move the check to there anyway, since ubuntu-18 has an old clang-format that doesn't support `--dry-run`.  Should I move the other check there too and rename the CI action?

Reply:
Yes, please do that.